src/lib/util.js

```diff
@@ -221,54 +221,53 @@ async function processSpecification(spec, callback, args, counter) {
             await page.goto(spec.url, options);
         }
 
-        // If the spec is a multi-page spec and contains a "Single page" link,
-        // extract the URL of the single page and load it instead
-        const singlePageUrl = await page.$$eval('body .head dl a[href]', links => {
-            const link = links.find(link => {
-                const text = (link.textContent || '').toLowerCase();
-                return text.includes('single page') ||
-                    text.includes('single file') ||
-                    text.includes('single-page') ||
-                    text.includes('one-page');
-            });
-            if (link) {
-                const url = new URL(link.getAttribute('href'), document.baseURI);
-                return url.href;
-            }
-            else {
-                return null;
-            }
-        });
-        if (singlePageUrl && (singlePageUrl !== spec.url) && (singlePageUrl !== page.url())) {
-            return processSpecification(singlePageUrl, callback, args, counter + 1);
-        }
-
         // Handle remaining multi-page specs manually, merging all subpages
         // into the main page to create a single-page spec.
+        // TODO: Move the page extraction logic to browser-specs
         let multiPagesRules = {
-            'https://www.w3.org/TR/CSS2/': '.quick.toc .tocxref',
-            'https://www.w3.org/TR/CSS22/': '#toc .tocxref',
-            'https://drafts.csswg.org/css2/': '#toc .tocline1 > .tocxref'
+            'https://html.spec.whatwg.org/multipage/': '#contents + .toc a[href]',
+            'https://www.w3.org/TR/CSS2/': '.quick.toc a[href]',
+            'https://www.w3.org/TR/CSS22/': '#toc a[href]',
+            'https://drafts.csswg.org/css2/': '#toc a[href]',
+            'https://www.w3.org/TR/SVG2/': '#toc a[href]',
+            'https://svgwg.org/svg2-draft/': '#toc a[href]',
         };
         if (multiPagesRules[page.url()]) {
-            let urls = await page.$$eval(multiPagesRules[page.url()], links =>
-                links.map(link => (new URL(link.getAttribute('href'), link.ownerDocument.baseURI)).toString()));
+            // Extract URLs that appear in the TOC and drop duplicates
+            const tocUrls = await page.$$eval(multiPagesRules[page.url()],
+                links => links.map(link =>
+                    (new URL(link.getAttribute('href'), link.ownerDocument.baseURI)).toString()));
+            const pageUrls = tocUrls
```

Comment:
maybe use a set to convey more directly the uniqueness semantics (but I don't feel strongly at all about it)

Reply:
True. Done!